components/campaign/CampaignChaosBagView/index.tsx

```diff
@@ -0,0 +1,363 @@
+  }
+
+  renderChaosToken = () => {
+    const { chaosBagResults } = this.props;
+    const iconKey = chaosBagResults.drawnTokens[chaosBagResults.drawnTokens.length - 1] || undefined;
+
+    return (
+      <TouchableOpacity onPress={this._handleDrawTokenPressed}>
+        <ChaosToken iconKey={iconKey} />
+      </TouchableOpacity>
+    );
+  }
+
+  renderDrawnTokens() {
+    const { chaosBagResults } = this.props;
+
+    const drawnTokens = chaosBagResults.drawnTokens;
+
+    if (drawnTokens.length > 1) {
+      return drawnTokens.slice(0, drawnTokens.length - 1).map(function(token, index) {
+        return (
+          <ChaosToken key={index} iconKey={token} small />
+        );
+      });
+    }
+
+    return <Text style={[styles.drawTokenText, typography.text]}>{ t`Tap token to draw` }</Text>;
+  }
+
+  renderDrawButton = () => {
+    const { chaosBagResults } = this.props;
+
+    if (chaosBagResults.drawnTokens.length > 0) {
+      return <Button title={t`Add and Draw Again`} onPress={this._handleAddAndDrawAgainPressed} />;
+    }
+  }
+
+  renderClearButton = () => {
+    const { chaosBagResults } = this.props;
+
+    if (chaosBagResults.drawnTokens.length > 1) {
+      return <Button title={t`Clear Tokens`} onPress={this._handleClearTokensPressed} />;
+    }
+  }
+
+  render() {
+    const { chaosBagResults } = this.props;
+
+    return (
+      <View style={styles.container}>
+        <View style={styles.containerTop}>
+          <View style={styles.chaosTokenView}>
+            { this.renderChaosToken() }
+          </View>
+          <View style={styles.drawButtonView}>
+            { this.renderDrawButton() }
+          </View>
+        </View>
+        <ScrollView style={styles.containerBottom}>
+          <View style={styles.header}>
+            <Text style={typography.text}>{ t`Drawn` }</Text>
+            <Text style={typography.small}>{ t`Total` } ({ chaosBagResults.totalDrawnTokens })</Text>
+          </View>
+          <View style={styles.container}>
+            <View style={styles.drawnTokenRow}>
+              { this.renderDrawnTokens() }
+            </View>
+            <View>
+              { this.renderClearButton() }
+            </View>
+          </View>
+          <View style={styles.header}>
+            <Text style={typography.text}>{ t`Sealed Tokens` }</Text>
+          </View>
+          <View style={styles.container}>
+            <Button title={t`Seal Token`} onPress={this._handleSealTokenPressed} />
+          </View>
+        </ScrollView>
+      </View>
+    );
+  }
+}
+
+const EMPTY_CHAOS_BAG = {
+  drawnTokens: [],
+  sealedTokens: [],
+  totalDrawnTokens: 0,
+};
+
+function mapStateToProps(
+  state: AppState,
+  props: NavigationProps & CampaignChaosBagProps
+): ReduxProps {
+  const campaign = getCampaign(state, props.campaignId);
+  return {
+    chaosBag: campaign ? campaign.chaosBag : {},
+    chaosBagResults: campaign ? campaign.chaosBagResults : EMPTY_CHAOS_BAG,
```

Comment:
Make this `chaosBagResults: (campaign && campaign.chaosBagResults) || EMPTY_CHAOS_BAG,` so it handles old campaigns where chaosBagResults isn't set.